src/StatementProvider.js

```diff
@@ -0,0 +1,73 @@
+( function ( wb, $ ) {
+	'use strict';
+
+	/**
+	 * Combination of a claim, a rank and references.
+	 * @mixin wikibase.datamodel.StatementProvider
+	 *
+	 * @param {wikibase.datamodel.StatementGroupSet|null} [statementGroupSet=new wikibase.datamodel.StatementGroupSet()]
+	 * @license GPL-2.0+
+	 */
+	var SELF = wb.datamodel.StatementProvider = function WbDataModelStatementProvider( statementGroupSet ) {
+		statementGroupSet = statementGroupSet || new wb.datamodel.StatementGroupSet();
+
+		if( !( statementGroupSet instanceof wb.datamodel.StatementGroupSet ) ) {
+			throw new Error( 'Given argument `statementGroupSet` is not instance of `wb.datamodel.StatementGroupSet`' );
+		}
+		this._statementGroupSet = statementGroupSet;
+	};
+
+	/**
+	 * @class wikibase.datamodel.Statement
+	 */
+	$.extend( SELF.prototype, {
+		/**
+		 * @property {wikibase.datamodel.StatementGroupSet}
+		 * @protected
+		 */
+		_statementGroupSet: null,
+
+		/**
+		 * @return {wikibase.datamodel.StatementGroupSet}
+		 */
+		getStatements: function () {
+			return this._statementGroupSet;
+		},
+
+		/**
+		 * @param {wikibase.datamodel.Statement} statement
+		 */
+		addStatement: function ( statement ) {
+			this._statementGroupSet.addStatement( statement );
+		},
+
+		/**
+		 * @param {wikibase.datamodel.Statement} statement
+		 */
+		removeStatement: function ( statement ) {
+			this._statementGroupSet.removeStatement( statement );
+		},
+
+		getStatementByGuid: function (guid) {
```

Comment:
I had an other look at the code and I suggest to move this convenience function up one level, into the StatementGroupSet class. Look at the convenience function `Entity.removeStatement()` directly above. It is nothing but a shortcut for `Entity.getStatements().removeStatement()`. The same should be done for the new getStatementByGuid().

The convenience shortcuts might not even be needed. The only method that's really needed in this mixin is getStatements.

But it gets more mysterious. Search for "addStatement" in our JS code. It's neither used nor declared. All these methods that are called on the StatementGroupSet class don't exist! What happened here?